pkgs/by-name/oa/oama/package.nix

```
@@ -3,17 +3,38 @@
   haskellPackages,
   lib,
   stdenv,
+  coreutils,
+  libsecret,
+  gnupg,
+  makeWrapper,
+  withLibsecret ? true, # default oama config uses libsecret
+  withGpg ? false,
 }:
 let
   inherit (haskell.lib.compose) overrideCabal justStaticExecutables;
 
   overrides = {
-    patches = [ ./0001-Downgrade-cabal-version-for-ghc-9.6-compat.patch ];
+    patches = [
+      ./0001-Downgrade-cabal-version-for-ghc-9.6-compat.patch
+    ];
     description = "OAuth credential MAnager";
     homepage = "https://github.com/pdobsan/oama";
     maintainers = with lib.maintainers; [ aidalgol ];
 
     passthru.updateScript = ./update.sh;
+
+    buildDepends = [
+      makeWrapper
```

You could try `makeBinaryWrapper`, which creates a binary executable instead of a shell script wrapper.

```suggestion
      makeBinaryWrapper
```

---

Thanks for the suggestion, it seems to work fine!